parsers/US_CA.py

```diff
@@ -248,14 +249,81 @@ def _get_oasis_data(session: Session, target_url: str) -> pd.DataFrame:
     return df
 
 
+@refetch_frequency(timedelta(days=7))
+def fetch_generation_forecast(
+    zone_key: ZoneKey = ZoneKey("US-CAL-CISO"),
+    session: Session | None = None,
+    target_datetime: datetime | None = None,
+    logger: Logger = getLogger(__name__),
+) -> list[dict[str, Any]]:
+    """Requests the total load forecast 7 days ahead (in MW) for a given date in hourly intervals."""
+    session = session or Session()
+
+    # Interval of time
+    if target_datetime is None:
+        target_datetime = datetime.now(
+            tz=timezone.utc
+        )  # .replace(hour=0, minute=0, second=0, microsecond=0) # TODO: is it necessary to replace the time?
+    target_datetime_gmt = target_datetime
+    GMT_URL_SUFFIX = "-0000"
+    END_OFFSET = timedelta(days=7)
+    startdatetime = target_datetime_gmt.strftime("%Y%m%dT%H:%M") + GMT_URL_SUFFIX
+    enddatetime = (target_datetime_gmt + END_OFFSET).strftime(
+        "%Y%m%dT%H:%M"
+    ) + GMT_URL_SUFFIX
```

Comment:
Question:
Does the endpoint support fully formated ISO dates?
If so you can just use datetime.isoformat() instead of adding the URL suffix and doing strftime.